draft-issuance.md

```diff
@@ -0,0 +1,195 @@
+
+Additionally, the current Bitcoin-style issuance does not take into account the current balance of `ZsfBalanceAfter(h)`. If [#draft-zsf]_ were to activate without a change to the issuance mechanism, then some funds would never be disbursed after they are deposited back into the ZSF.
+
+In summary, by introducing a smoother emissions curve, we:
+- maintain the economic viability of Zcash
+- provide predictability of the issuance rate, allowing only miniscule changes over short time ranges
+- enhance Zcash's stability as the network evolves.
+
+# Requirements
+
+Smoothing the issuance curve is possible using an exponential decay formula that
+satisfies the following requirements:
+
+## Issuance Requirements
+
+1. The issuance can be summarised into a reasonably simple explanation
+2. Block subsidies approximate a continuous function
+3. If there are funds in the ZSF, then the block subsidy must be non-zero, preventing any final “unmined” zatoshis
+4. For any 4 year period, all paid out block subsidies are approximately equal to half of the ZSF at the beginning of that 4 year period, if there are no deposits into the ZSF during those 4 years
+TODO daira: add a requirement that makes the initial total issuance match the previous total issuance  
+5. This functionality MUST be introduced as part of a network upgrade
+
+
+# Specification
+
+## Goals
+
+We want to decrease the short-term impact of the deployment of this ZIP on block reward recipients, and minimise the potential reputational risk to Zcash of changing the block reward amount.
+
+## Constants
+
+Define constants:
+
+“`BLOCK_SUBSIDY_FRACTION`” = 4126 / 10,000,000,000 or `0.0000004126`
+
+"`DEPLOYMENT_BLOCK_HEIGHT`" = 2726400
+
+## Issuance Calculation
+
+At the `DEPLOYMENT_BLOCK_HEIGHT`, nodes should switch from the current issuance calculation, to the following:
+
+Given the block height `h` define a function **BlockSubsidy(h)**, such that:
+
+**BlockSubsidy(h)** = Block subsidy for a given `h`, that satisfies above requirements.
+
+Using an exponential decay function for **BlockSubsidy** satisfies requirements **R1** and **R2** above:
+
+`BlockSubsidy(h) = BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1)`
+
+Finally, to satisfy **R3** above we always round up to the next zatoshi.
+
+`BlockSubsidy(h) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1))`
+
+## Deployment
+
+TBD
+
+# Rationale
+
+## `BLOCK_SUBSIDY_FRACTION`
+
+Let `IntendedZSFFractionRemainingAfterFourYears` = 0.5.
+
+The value `4126 / 10_000_000_000` satisfies the approximation within +0.002%:
+
+`(1 - BLOCK_SUBSIDY_FRACTION)^PostBlossomHalvingInterval ≈ IntendedZSFFractionRemainingAfterFourYears`
+
+Meaning after a period of 4 years around half of `ZSF_BALANCE` will be paid out
+as block subsidies, thus satisfying **R4**.
+
+TODO for ZIP owners: How many ZEC per day?
```

Comment:
@tomekpiotrowski as part of your simulator, it would be helpful to calculate a total block subsidy for the next halving, so we can compare it to the total block subsidy if we didn't make this change.

The ZSF subsidy is the subsidy for the first `1_680_000` (`PostBlossomHalvingInterval`) blocks.
The non-ZSF subsidy is over the same block range, but it is a constant `1.5625` ZEC per block. (This is equal to half the current `3.125` ZEC subsidy per block.)

Optional: It might also be helpful to make `BLOCKS_PER_YEAR` equal to `1_680_000/4` to avoid confusion.

Reply:
@teor2345 I've updated my simulator code: https://github.com/eigerco/zsf-simulator#output with the comparison of subsidies paid out between halvings. I'll update the ZIP itself soon as well.